rfcs/2019-04-01-externally-changed-file.md

```diff
@@ -0,0 +1,53 @@
+# RFC: Acting upon an externally changed file
+
+Right now Xi only has two mechanisms to handle externally changed files:
+
+* Blocking saving, instead requiring the user to save the file elsewhere
+* Automatically reloading the file if no changes have been done
+
+Blocking saving is a bit annoying, especially if the user _does_ want to overwrite
+the file (e.g. when a git rebase has modified a file's modtime).
+
+## Summary
+
+For this to work we have to change multiple thing:
+
+- Add a new notification ('file_externally_changed')
+	- Frontends may use this to display a 'File has been changed, want to reload?' dialog
+- Add a way for frontends to force reload the file to implement the previously mentioned dialog
+- Add a way for frontends to force save the file, e.g. via a 'force' param to 'save'
```

Comment:
I agree that these additions should be enough to get us what we want. 👍 

We probably only want to send this `file_changed` notification if the changes can't be auto-applied, although I could imagine making this a user preference at some point.

Reply:
Oh, what kind of changes are auto-applied?

Reply:
Any time a file changes on disk and there are no unsaved changed in the editor, we reload the changed file automatically.

Reply:
Alright, thanks for the info :)